src/Symfony/Component/HttpKernel/DependencyInjection/ControllerArgumentValueResolverPass.php

```diff
@@ -37,7 +38,21 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        $resolvers = $this->findAndSortTaggedServices('controller.argument_value_resolver', $container);
+        $definitions = $container->getDefinitions();
+        $namedResolvers = $this->findAndSortTaggedServices(new TaggedIteratorArgument('controller.pinned_value_resolver', 'name', needsIndexes: true), $container);
+        $resolvers = $this->findAndSortTaggedServices(new TaggedIteratorArgument('controller.argument_value_resolver', 'name', needsIndexes: true), $container);
+
+        foreach ($resolvers + $namedResolvers as $name => $resolverReference) {
```

Comment:
not sure about the ` + $namedResolvers`: it should be moved to the next foreach to me (the one that decorates)

Reply:
Indeed, misread your diff